client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -316,11 +316,23 @@ const PatternAssembler = ( {
 	) => {
 		if ( selectedPattern ) {
 			// Inject the selected pattern category or the first category
-			// because it's used in tracks and as pattern name in the list
+			// to be used in tracks and as selected pattern name.
 			const [ firstCategory ] = Object.keys( selectedPattern.categories );
-			selectedPattern.category = categories.find(
-				( { name } ) => name === ( selectedCategory || firstCategory )
-			);
+			selectedPattern.category = categories.find( ( { name } ) => {
+				if ( selectedCategory === CATEGORY_ALL_SLUG ) {
+					return name === firstCategory;
+				}
+				return name === ( selectedCategory || firstCategory );
```

Comment:
I just found out there is an issue here before. If the first category is in front of the selected category, we will use the first category instead of the selected one.

Reply:
Just to confirm, it is now solved in this version, right?

Could you help me understand the issue? I don't get what you mean by "If the first category is in front of the selected category".

Reply:
For example, 

```js
const categories = [ 'text', 'blog' ];
const selectedCategory = 'blog';
const selectedPattern = {
  categories: {
    'text': ...,
    'blog': ...,
  }
}

const [ firstCategory ] = Object.keys( selectedPattern.categories );
// firstCategory: text
```

Next, when we loop through the `categories`, we will get `text` instead of the blog because the `firstCategory` will be matched at the beginning.

Reply:
Do you mean that will be matched in this line `name === ( selectedCategory || firstCategory )`?

Reply:
Ooops...I found I misread something...you can ignore what I said 🤦‍♂️